frontend/crd/src/app/milestones-page/milestones/milestones.component.ts

```diff
@@ -28,10 +32,29 @@ export class MilestonesComponent implements OnInit, OnDestroy {
       .subscribe((milestones: Milestone[]) => {
         this.yearLevels.forEach((yearLevel) => this.milestonesMap.set(yearLevel, new Array<Milestone>()));
         milestones.forEach((milestone) => this.milestonesMap.get(milestone.yearLevel)?.push(milestone));
-    });
+      });
   }
 
-  milestonesMap: Map<string, Array<Milestone>> = new Map()
+  openTask(task: any) {
+    console.log(task)
+    const dialogConfig = new MatDialogConfig();
+    // The user can't close the dialog by clicking outside its body
+    dialogConfig.disableClose = true;
+    dialogConfig.id = "modal-component";
+    dialogConfig.height = "80%";
+    dialogConfig.width = "60%";
+    dialogConfig.data = {
+      // name: task.name,
+      // title: "TASK",
+      // description: task.description,
+      // actionButtonText: "Complete",
+      // productId: task.id
+      task: task,
+      actionButtonText: "TESTING"
```

Comment:
what is this testing button for? if we're not going to put in something here just remove it